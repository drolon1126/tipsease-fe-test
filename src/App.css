@import url("https://fonts.googleapis.com/css?family=Varela+Round&display=swap");
@import url('https://fonts.googleapis.com/css?family=Ubuntu:400,500&display=swap');


* {
  box-sizing: border-box;
}

.App {
  text-align: center;
  width: 82%;
  height: 100vh;
  margin: 70px auto 0 auto;
  background: linear-gradient(#7f8688, #f2f7f4);
}

html,
body {
  margin: 0;
  padding: 0;
  font-size: 62.5%;
  background: #e8e8e8;
  width: 100%;
  height: 100%;
}

main {
  width : 85%;
  margin: auto;
}

header {
  margin: 0 auto -1em auto;
}

.signUpPage {
  height: 100vh;
}

.formPageImg {
  position: fixed; 
  top: 70px; 
  left: 0; 
	
  /* Preserve aspet ratio */
  width: 100%;
  height: 100%;

}

.formInp {
  color: #EDF2F0;
}

.active {
  color: white;
}

.MuiGrid-container {
  justify-content: center;
}

.tip-button-container {
  margin-bottom: 3rem;
}

.active-cta {
  color: #f7eb95;
}
#email,
#username,
#password,
#passwordConfirmation,
.textInputField {
  margin-bottom   : -2em;
  background-color: #EDF2F0;
  border          : 2px solid black;
  border-radius   : 10px;
  padding         : 6px;
}

.formButt {
  background   : none;
  color        : #f3e367;
  border       : 1px solid #f3e367;
  padding      : 1.2rem 2.4rem;
  font-size    : 2rem;
  font-family  : "Ubuntu", sans-serif;
  border-radius: 10px;
  transition   : all 250ms ease-out;
}

.formButt:hover {
  cursor    : pointer;
  background: #f3e367;
  color     : black;
}

.tipButton {
  background   : none;
  color        : #f3e367;
  border       : 1px solid #f3e367;
  padding      : 1.2rem 2.4rem;
  font-size    : 2rem;
  font-family  : "Ubuntu", sans-serif;
  border-radius: 10px;
  transition   : all 250ms ease-out;
}

.tipButton:hover {
  cursor    : pointer;
  background: #f3e367;
  color     : black;
}

.tipButton svg {
  margin-right: 0.6rem;
}

.wrkNameImg {
  width: 150%;
}