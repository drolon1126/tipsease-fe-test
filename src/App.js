
import React, { useState, useEffect, useContext } from "react";
import { Route, NavLink } from "react-router-dom";
import axios from 'axios'
import PrivateRoute from "./Components/SecretData/PrivateRoute";
import Login from "./Components/Login/Login";

import Home from "./Components/LandingPage/Home";
import { TokenContext, WorkersListContext, FilterContext } from "./Components/Context/Contexts";
import SignUp from "./Components/SignUp/SignUp";
import styled from "styled-components";
import Swal from "sweetalert2";
import TipForm from "./Components/LandingPage/Workers/TipForm";
import TipModal from "./Components/LandingPage/Workers/TipModal";
import "./App.css";
import CreateWorker from './Components/LandingPage/CreateWorker/CreateWorker'



const NavbarContainer = styled.div`
  width: 100%;
  height: 70px;
  background: rgba(0, 0, 0, 0.9);
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 0.5rem 4.8rem;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 100;

  @media (max-width: 800px) {
    padding: 0.5rem 2.4rem;
    background: black;
  }
`;

const NavLogo = styled.div`
  font-size: 2.9rem;
  color: #f3e367;
  margin-right: auto;
  font-family: "Ubuntu", sans-serif;
  font-weight: 500;

  @media (max-width: 800px) {
    font-size: 2.2rem;
  }
`;

const NavItems = styled.div`
  width: 40%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  font-family: "Varela Round", sans-serif;


  @media (max-width: 1100px) {
    width: 55%;
  }

  @media (max-width: 800px) {
    width: 70%;
  }
`;

const NavItem = styled(NavLink)`
  color: #c9c9c9;
  text-transform: uppercase;
  transition: color 200ms ease-out;
  padding: 0.5rem;
  font-size: 1.6rem;
  text-decoration: none;
  :hover {
    color: white;
    cursor: pointer;
  }
  @media (max-width: 800px) {
    width: 100%;
    margin: 1rem 0;
    font-size: 1.4rem;
    text-align: center;
  }
  @media (max-width: 600px) {
    font-size: 1rem;
  }
`;

const NavCta = styled(NavItem)`
  color: #f3e367;
`;


const Logout = styled.button`
  background: none;
  color: #f3e367;
  border: 2px solid #f3e367;
  padding: 0.8rem 1.5rem;
  font-size: 1.5rem;
  border-radius: 10px;
  transition: all 200ms ease-out;
  text-decoration: none;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-family: 'Ubuntu', sans-serif;
  font-weight: 500;
  :hover {
    cursor: pointer;
    background: #f3e367;
    color: black;
  }
  :focus {
      outline: none;
  }
  @media (max-width: 800px) {
    font-size: 1.3rem;
    padding: 0.8rem 1.2rem;
    border-width: 1px;
  }
  @media (max-width: 600px) {
    font-size: 1rem;
    padding: 0.5rem 1rem;
    border-width: 1px;
  }
`;
function App() {
  const [token, setToken] = useState(false);
  const [workerList, setWorkerList] = useState([])
  const [filter, setFilter] = useState("all");

  useEffect(() => {
    setToken(localStorage.getItem("token"));
    console.log("test");
  }, [token]);
  useEffect(() => {
    axios.get(`https://tipsease-be-test.herokuapp.com/api/tippees`)
      .then(apiObject => {
        setWorkerList(apiObject.data)
      })
      .catch(err => {
        console.log("Error:", err);
      })
  }, []);

  // console.log('Token u there brother?: ', token);
  return (
    <div style={{height:'100%'}}>
      <NavbarContainer>
        <NavLogo>tipsEase</NavLogo>
        <NavItems className="nav-items">
          <NavItem exact to="/">Home</NavItem>
          <NavItem to="/About">About</NavItem>
          <NavItem to="/Team">Team</NavItem>
          <NavCta to="/SignUp" activeClassName="active-cta">
            Sign Up
          </NavCta>
          {token === null ? (
            <NavCta to="/login" activeClassName="active-cta">
              Login
            </NavCta>
          ) : (
              <NavCta to="/login" activeClassName="active-cta">
                <Logout
                  className="btn"
                  type="submit"
                  onClick={() => {
                    localStorage.removeItem("token");
                    setToken();
                  }}
                >
                  Logout
              </Logout>
              </NavCta>
            )}
        </NavItems>
      </NavbarContainer>

      <div style={{height:'100%'}}>
        {/* export const TokenContext = createContext(); */}
        <TokenContext.Provider value={{ token, setToken }}>
          <WorkersListContext.Provider value={{ workerList, setWorkerList }}>
            <FilterContext.Provider value={{ filter, setFilter }}>
              <PrivateRoute
                exact
                path="/"
                component={Home}
              //  token ={token}
              />
              <Route exact path="/About" />
              <Route exact path="/Team" />
              <Route
                exact
                path="/login"
                render={props => <Login {...props} setToken={setToken} />}
              />
              <Route exact path="/signup" component={SignUp} />
              <Route path="/TipForm" component={TipForm} />
              <Route exact path="/CreateWorker" component={CreateWorker} />
            </FilterContext.Provider>
          </WorkersListContext.Provider>
        </TokenContext.Provider>
      </div>
    </div>
  );
  // const handleSubmit =() =>{
  //   Swal.fire({
  //     position: 'center',
  //     type: 'success',
  //     title: 'Logging out!',
  //     showConfirmButton: false,
  //     timer: 2500
  //   })
  // }
}

export default App;

