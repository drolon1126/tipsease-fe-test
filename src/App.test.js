
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import {BrowserRouter} from "react-router-dom"

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><App /></BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});


  it('should match snapshot', () => {
    const addingtesting = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();

    expect(addingtesting).toMatchSnapshot();
  });