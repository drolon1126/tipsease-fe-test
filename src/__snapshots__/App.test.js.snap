// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <header>
    <h1>
      TipsEase
    </h1>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/SignUp"
      onClick={[Function]}
    >
      Sign Up
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </header>
  <div>
    <form
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <label
        htmlFor="username"
      >
        username
      </label>
      <input
        autoComplete="off"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="username"
        type="username"
        value=""
      />
      <h5
        className="sc-bdVaJa kncjvf"
      />
      <label
        htmlFor="password"
      >
        password
      </label>
      <input
        autoComplete="off"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
      <h5
        className="sc-bdVaJa kncjvf"
      />
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
